sprocs/grading_jobs_process_external.sql

```diff
@@ -57,28 +58,43 @@ BEGIN
         grading_received_at = received_time,
         grading_started_at = start_time,
         grading_finished_at = finish_time,
+        gradable = new_gradable,
         score = grading_jobs_process_external.score,
         correct = (grading_jobs_process_external.score >= 1.0),
         feedback = grading_jobs_process_external.feedback
     WHERE id = grading_job_id
     RETURNING *
     INTO grading_job;
 
-    UPDATE submissions
-    SET
-        graded_at = grading_job.graded_at,
-        score = grading_job.score,
-        correct = grading_job.correct,
-        feedback = grading_job.feedback
-    WHERE id = grading_job.submission_id;
+    IF new_gradable = FALSE THEN
+        UPDATE submissions
+        SET
+            gradable = FALSE,
+            feedback = grading_jobs_process_external.feedback
+        WHERE id = grading_job.submission_id;
 
-    -- ######################################################################
-    -- update all parent objects
+        IF assessment_instance_id IS NOT NULL THEN
+            UPDATE instance_questions
+            SET status = 'invalid'::enum_instance_question_status
+            WHERE id = instance_question_id;
+        END IF;
+    ELSE
+        UPDATE submissions
+        SET
```

Comment:
Should we set `gradable = TRUE` here?

Reply:
Yep, if it is not already gradable here then this code won't execute.

Reply:
Excellent.